src/main/java/org/elasticsearch/search/aggregations/metrics/scripted/ScriptedMetricBuilder.java

```diff
@@ -187,20 +299,20 @@ protected void internalXContent(XContentBuilder builder, Params builderParams) t
             builder.map(reduceParams);
         }
         
-        if (initScript != null) {
-            builder.field(ScriptedMetricParser.INIT_SCRIPT, initScript);
+        if (initScriptString != null) {
+            builder.field(ScriptedMetricParser.INIT_SCRIPT, initScriptString);
         }
         
-        if (mapScript != null) {
-            builder.field(ScriptedMetricParser.MAP_SCRIPT, mapScript);
+        if (mapScriptString != null) {
+            builder.field(ScriptedMetricParser.MAP_SCRIPT, mapScriptString);
         }
         
-        if (combineScript != null) {
-            builder.field(ScriptedMetricParser.COMBINE_SCRIPT, combineScript);
+        if (combineScriptString != null) {
+            builder.field(ScriptedMetricParser.COMBINE_SCRIPT, combineScriptString);
         }
         
-        if (reduceScript != null) {
```

Comment:
Any validation checks needed here for a mix of old-style AND new-style params e.g. specifying script as a string AND an ID AND a Script object?
Or is this handled post-builder by the request parser?